### PR TITLE
fix(postgresql): buildDataType NPE for NUMERIC when precision null but scale set

### LINKED ISSUE #2438: bug(postgresql): buildDataType NPEs for NUMERIC when precision null but scale set
## Description

`PostgreSQLColumnTypeEnum.buildDataType` for DECIMAL/NUMERIC: when `columnSize` is null but `decimalDigits` is set, control falls to the else branch which does `column.getColumnSize() + ","` — unboxing a null `Integer` → NPE. Affects the PostgreSQL family that inherits this enum (GaussDB, OpenGauss, CockroachDB, Redshift).

## Location

`chat2db-community-plugins/chat2db-community-postgresql/.../enums/type/PostgreSQLColumnTypeEnum.java:405-413`

## Impact

DDL generation NPEs for a NUMERIC/DECIMAL column whose scale is set but precision is null.

## Suggested fix

Restructure the branches so a null `columnSize` returns the bare type (precision is required to specify scale), instead of unboxing null.

## Related existing

None.

### PR DESCRIPTION
**Pull request by Aias00**
## What
When `columnSize` was null but `decimalDigits` was set, the DECIMAL/NUMERIC branch fell to the else and unboxed null `columnSize` → NPE.

## Location
`chat2db-community-plugins/chat2db-community-postgresql/.../enums/type/PostgreSQLColumnTypeEnum.java:405-413` (affects the PG family: GaussDB/OpenGauss/CockroachDB/Redshift).

## Fix
Restructure so a null precision returns the bare type (precision is required to specify scale). Preserves: both null → bare; precision only → `(p)`; precision+scale → `(p,s)`.

Fixes #2438

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: chat2db-community-server/chat2db-community-plugins/chat2db-community-postgresql/src/main/java/ai/chat2db/plugin/postgresql/enums/type/PostgreSQLColumnTypeEnum.java [hunk 1]
```diff
@@ -403,14 +403,16 @@ private String buildDataType(TableColumn column, PostgreSQLColumnTypeEnum type)
...
+            if (column.getColumnSize() == null) {
                 return columnType;
             }
-            if (column.getColumnSize() != null && column.getDecimalDigits() == null) {
+            if (column.getDecimalDigits() == null) {
                 return StringUtils.join(columnType, "(", column.getColumnSize() + ")");
-            } else {
-                return StringUtils.join(columnType, "(", column.getColumnSize() + "," + column.getDecimalDigits() + ")");
             }
+            return StringUtils.join(columnType, "(", column.getColumnSize() + "," + column.getDecimalDigits() + ")");
```
[the remark this answers is not in the document]
Comment (openai0229) [reply chat2db-community-server/chat2db-community-plugins/chat2db-community-postgresql/src/main/java/ai/chat2db/plugin/postgresql/enums/type/PostgreSQLColumnTypeEnum.java]: Addressed in af6ea94e: precision and scale are now passed as separate StringUtils.join arguments, and the focused numeric tests still pass.

File: chat2db-community-server/chat2db-community-plugins/chat2db-community-postgresql/src/main/java/ai/chat2db/plugin/postgresql/enums/type/PostgreSQLColumnTypeEnum.java [hunk 2]
```diff
@@ -403,14 +403,16 @@ private String buildDataType(TableColumn column, PostgreSQLColumnTypeEnum type)
         }
 
         if (Arrays.asList(DECIMAL, NUMERIC).contains(type)) {
-            if (column.getColumnSize() == null && column.getDecimalDigits() == null) {
+            // precision is required to specify scale; without it, emit the bare type
+            // (previously a null precision with a set scale fell through and NPE'd on
+            // null Integer unboxing). Affects the PG family (GaussDB/OpenGauss/CockroachDB/Redshift).
+            if (column.getColumnSize() == null) {
                 return columnType;
```
[the remark this answers is not in the document]
Comment (openai0229) [reply chat2db-community-server/chat2db-community-plugins/chat2db-community-postgresql/src/main/java/ai/chat2db/plugin/postgresql/enums/type/PostgreSQLColumnTypeEnum.java:410]: Addressed in fb72f13a: the new PostgreSQLColumnTypeEnumNumericTest covers NUMERIC and DECIMAL with null precision plus a set scale, as well as precision-only and precision-plus-scale formatting.

**Review by openai0229 — APPROVED** [review]
The null-precision path now emits a valid bare NUMERIC/DECIMAL type, formatting is consistent, and focused PostgreSQL tests cover the edge case plus normal precision/scale forms.
